docker/utils/utils.py

```diff
@@ -1028,6 +1037,17 @@ def create_container_config(
             volumes_dict[vol] = {}
         volumes = volumes_dict
 
+    if isinstance(healthcheck, dict):
+        healthcheck = {
+            'Test': [
+                'CMD-SHELL',
+                healthcheck.get('command')
+            ],
+            'Interval': duration_from_string(healthcheck.get('interval')) if healthcheck.has_key('interval') else None,
+            'Timeout': duration_from_string(healthcheck.get('timeout')) if healthcheck.has_key('timeout') else None,
+            'Retries': healthcheck.get('retries')
+        }
+
```

Comment:
We need to raise an error if `healthcheck` isn't the right type. You can use `host_config_value_error`.